### PR TITLE
Publish UX

### PR DESCRIPTION
**Pull request by akinwale**

### CODE REVIEW

File: src/page/publish/view.js [hunk 1]
```diff
@@ -695,38 +756,46 @@ class PublishPage extends React.PureComponent {
...
-      licenseUrl: LICENSES.CC_LICENSES.reduce((value, item) => {
-        if (typeof value === 'object') {
-          value = '';
-        }
-        if (license === item.value) {
-          value = item.url;
-        }
-        return value;
-      }),
+    const licenseUrl = LICENSES.CC_LICENSES.reduce((value, item) => {
```
Comment (kauffj) [inline src/page/publish/view.js]: are you sure this does what you want it to do? won't the end state of `value` depend on the order of items in the array?

File: src/page/publish/view.js [hunk 2]
```diff
@@ -186,26 +193,44 @@ class PublishPage extends React.PureComponent {
...
         }
       }
+
+      if (!isEditMode && hasFormState) {
+        this.loadPendingFormState();
+        if (vanityUrlSet) {
+          // replace name with the specified vanity URL if there was one in the pending state
+          this.setState({ name: this.state.vanityUrl });
+        }
+        this.setState({ currentPhase: Constants.PHASE_DETAILS });
```
Comment (kauffj) [inline src/page/publish/view.js]: This code along with some of the code in `componentWillReceiveProps` makes me think there is a cleaner way of doing this, but this is a non-blocking objection and something we should discuss on upcoming 1-1 or standup.

File: src/page/publish/view.js (hunk 1 above)
Comment (akinwale) [reply src/page/publish/view.js]: Yes. I replace value with an empty string for each of the items, and only assign the url string when there's a url match found.

File: src/page/publish/view.js (hunk 2 above)
Comment (akinwale) [reply src/page/publish/view.js]: The `onComponentFocused` method is pretty much a workaround for the bugs I was facing with the navigation component not calling `componentDidMount` every time the user opened a particular page. It's something I'd like to address properly in the future.

File: src/page/publish/view.js (hunk 1 above)
Comment (kauffj) [reply src/page/publish/view.js]:
```
const licenseUrl = LICENSES.CC_LICENSES.find(item => { return item.value === license)).url;
```

(but better so if one isn't found it actually works)

**Pull request merged**